openapi/components/schemas/CreditMemos/CreditMemoProperties.yaml

```
@@ -0,0 +1,176 @@
+id:
+  description: Credit memo ID.
+  readOnly: true
+  allOf:
+    - $ref: ../ResourceId.yaml
+invoiceId:
+  description: ID of the invoice which the credit memo is issued upon.
+  nullable: true
+  allOf:
+    - $ref: ../ResourceId.yaml
+customerId:
+  $ref: ../CustomerId.yaml
+number:
+  description: An auto-incrementing number based on the sequence of credit memos for any particular customer.
+  readOnly: true
+  type: integer
+allocations:
+  type: array
+  items:
+    type: object
+    properties:
+      transactions:
+        type: array
+        items:
+          type: object
+          properties:
+            transactionId:
+              description: ID of the transaction that the credit memo is allocated to.
+              allOf:
+                - $ref: ../ResourceId.yaml
+            amount:
+              description: |-
+                Amount of credit that is allocated from the credit memo to the transaction.
+                This value cannot exceed the unused amount of the credit memo or the transaction amount.
+                If the `amount` value is not supplied, the lesser of the following two values is used:
+                - The unused amount of the credit memo.
+                - The transaction amount.
+              type: number
+              format: double
+            currency:
+              readOnly: true
+              allOf:
+                - $ref: ../CurrencyCode.yaml
+            createdTime:
+              description: Credit memo allocation time.
+              readOnly: true
+              allOf:
+                - $ref: ../ServerTimestamp.yaml
+            updatedTime:
+              $ref: ../UpdatedTime.yaml
+items:
+  type: array
+  items:
+    type: object
+    required:
+      - unitPrice
+      - quantity
+    properties:
+      id:
+        description: Credit memo item ID.
+        readOnly: true
+        allOf:
+          - $ref: ../ResourceId.yaml
+      invoiceItemId:
+        description: An invoice item's ID that the credit item is referenced to.
+        nullable: true
+        allOf:
+          - $ref: ../ResourceId.yaml
+      description:
+        description: Credit memo item's description.
+        type: string
+      unitPrice:
+        description: Credit memo item's price.
+        type: number
+        format: double
+      quantity:
+        description: Credit memo item's quantity.
+        type: integer
+      price:
+        description: Credit memo item's total price.
+        type: number
+        format: double
+        readOnly: true
+      productId:
+        description: The related product's ID.
+        nullable: true
+        allOf:
+          - $ref: ../ResourceId.yaml
+      planId:
+        description: The related plan's ID.
```

nitpick: suggestion to avoid possessive ('s)
```suggestion
        description: ID of the related plan ID.
```